### ISSUE TITLE
Linux - ERROR Build failed - Unable to determine cause

### ISSUE THREAD

**Issue by solvingj** [codrsquad/portable-python#6]
This is a very strange issue.  All the major steps seem to have worked:

```
09:43:10  Python build finished successfully!
...
09:43:19  Successfully installed setuptools-62.4.0
```

Until the end, when portable python reports failure and returns a non-zero exit code:

```
09:43:21  ERROR Build failed:
```

I just can't find any trace of what actually failed.   I've attached the full log, maybe I'm missing something. 

I also looked at the code around the error reporting but didn't find any obvious clues to what happened.  Is there any way to re-run with more verbose output?

```
 INFO Cleaned 144 build artifacts (30 MB): 2to3 2to3-3.10 __phello__.foo.py __pycache__ idle3 idle3.10 pydoc3 pydoc3.10
09:43:20  INFO Symlink build/ppp-marker/3.10.4/lib/python3.10/config-3.10-x86_64-linux-gnu/libpython3.10.a <- build/ppp-marker/3.10.4/lib/libpython3.10.a
09:43:21  ERROR Build failed: Uses system libs: /lib/x86_64-linux-gnu/libbz2.so.1.0 /lib/x86_64-linux-gnu/libffi.so.7 /lib/x86_64-linux-gnu/libncursesw.so.6 /lib/x86_64-linux-gnu/libtinfo.so.6 /lib/x86_64-linux-gnu/libpanelw.so.6 /lib/x86_64-linux-gnu/libcrypto.so.1.1 /lib/x86_64-linux-gnu/liblzma.so.5 /lib/x86_64-linux-gnu/libsqlite3.so.0 /lib/x86_64-linux-gnu/libssl.so.1.1 /lib/x86_64-linux-gnu/libtk8.6.so /lib/x86_64-linux-gnu/libtcl8.6.so /lib/x86_64-linux-gnu/libXft.so.2 /lib/x86_64-linux-gnu/libfontconfig.so.1 /lib/x86_64-linux-gnu/libX11.so.6 /lib/x86_64-linux-gnu/libXss.so.1 /lib/x86_64-linux-gnu/libz.so.1 /lib/x86_64-linux-gnu/libfreetype.so.6 /lib/x86_64-linux-gnu/libXrender.so.1 /lib/x86_64-linux-gnu/libuuid.so.1 /lib/x86_64-linux-gnu/libxcb.so.1 /lib/x86_64-linux-gnu/libXext.so.6 /lib/x86_64-linux-gnu/libpng16.so.16 /lib/x86_64-linux-gnu/libXau.so.6 /lib/x86_64-linux-gnu/libXdmcp.so.6 /lib/x86_64-linux-gnu/libbsd.so.0 /lib/x86_64-linux-gnu/libreadline.so.8
09:43:21  INFO Compiling cpython took 7 minutes 35 seconds
09:43:21  INFO Overall compilation took 7 minutes 39 seconds
09:43:21          prefix  build/ppp-marker/3.10.4
09:43:21            _bz2  _bz2*.so bz2:1.0
09:43:21         _ctypes  1.1.0 _ctypes*.so ffi:7
09:43:21         _curses  2.2 _curses*.so ncursesw:6 tinfo:6
09:43:21            _dbm  *absent*
09:43:21        _decimal  1.70 _decimal*.so
09:43:21           _gdbm  *absent*
09:43:21           _lzma  _lzma*.so lzma:5
09:43:21        _sqlite3  3.31.1 _sqlite3*.so sqlite3:0
09:43:21            _ssl  OpenSSL 1.1.1f  31 Mar 2020 _ssl*.so ssl:1.1 crypto:1.1
09:43:21        _tkinter  8.6 _tkinter*.so tk8:8.6 tcl8:8.6 Xft:2 fontconfig:1 X11:6 Xss:1 z:1 freetype:6 Xrender:1 uuid:1 xcb:1 Xext:6 png16:16 Xau:6 Xdmcp:6 bsd:0
09:43:21           _uuid  _uuid*.so uuid:1
09:43:21             pip  22.1.2 lib/python3.10/site-packages/pip
09:43:21         pyexpat  2.4.7 pyexpat*.so
09:43:21        readline  8.0 readline*.so readline:8 tinfo:6
09:43:21      setuptools  *absent* /mnt/ws/workspace/dibr-python-tools/portable_python_builder/build/ppp-marker/3.10.4/lib/python3.10/distutils/core.py
09:43:21            zlib  1.2.11 zlib*.so z:1
09:43:21    libpython*.a  91 MB lib/libpython3.10.a lib/python3.10/config-3.10-x86_64-linux-gnu/libpython3.10.a
09:43:21   libpython*.so  -not used-
09:43:21    install size  162 MB
```

Full log:

[portable-python-ubuntu-failure.log](https://github.com/codrsquad/portable-python/files/8910043/portable-python-ubuntu-failure.log)

**Comment by zsimic** [comment]
You can get a more verbose output with: 
```
portable-python inspect --verbose build/ppp-marker/3.10.4
```

**Comment by zsimic** [comment]
Note: latest version corrects a bug when inspecting `setuptools`. Turns out if one imports `pip` first, it mucks with `setuptools`, and it ends up looking like setuptools isn't there...
The latest version of portable-python makes sure to import `pip` last (during `inspect`) so that side-effect doesn't happen

**Comment by zsimic** [comment]
Yes indeed, doc is lacking.
You can do `portable-python -c .../my-cfg.yml ...`

I will make it so that all the libs that are compiled in as `shared` (shown in `build-report`, or at beginning of a `build` run) don't require repeating that in a config (I missed that), in a future version

**Comment by solvingj** [comment]
In :  `./portable-python.yml`

Adding: 
```
linux:
  allowed-system-libs: /lib/.*
```
Solved my problem

**Comment by zsimic** [comment]
I've pushed a correction for this: https://github.com/codrsquad/portable-python/commit/201b4e59f2fe0ed11b7dd8bf1e35aaa0cf725768

With that, you shouldn't need to configure `allowed-system-libs` anymore. It was a bug I think.
Should be fixed in v1.4.5
